### PR TITLE
fix performance drop introduced by ElementFiniteDifferenceUtilities

### PR DESCRIPTION
**Pull request by armingeiser**
The number of function calls for the finite difference utility is reduced.

### CODE REVIEW

File: applications/StructuralMechanicsApplication/custom_response_functions/response_utilities/element_finite_difference_utility.cpp [hunk 1]
```diff
@@ -222,14 +201,11 @@ namespace Kratos
...
                 //compute derivative of RHS w.r.t. design variable with finite differences
-                noalias(rOutput) = (perturbed_mass_matrix - unperturbed_mass_matrix) / rPertubationSize;
+                noalias(rOutput) = (perturbed_mass_matrix - rMassMatrix) / rPertubationSize;
 
                  // unperturb the design variable
                 rNode.GetInitialPosition()[coord_dir] -= rPertubationSize;
                 rNode.Coordinates()[coord_dir] -= rPertubationSize;
-
-                //call one last time to make sure everything is as it was before TODO improve this..
-                rElement.CalculateMassMatrix(perturbed_mass_matrix, rCurrentProcessInfo);
```
Comment (armingeiser) [inline applications/StructuralMechanicsApplication/custom_response_functions/response_utilities/element_finite_difference_utility.cpp:232]: @MFusseder maybe you have an idea how to test this.
For the ShapeDerivatives it is seems not to be needed.

**Comment by dbaumgaertner** [comment]
refers to this issue @ #3082

**Comment by MFusseder** [comment]
My impression is that the last call of RHS, LHS, ... is not necessary for the elements we use currently in linear analysis. So I am fine that it is deleted within this PR. 
Nevertheless we should be very careful about this topic, especially when we move towards non-linear sensitivity analysis or if we use the tool with other elements. 
It is really hard to overview if some perturbed data is saved somewhere. Currently I have no idea on how to check this systematically. Of course one could call in a test stage RHS, LHS again and compare it with the initial one. But this does not ensure that somewhere else a 'wrong', perturbed value is used. I think here e.g. on some of the coordinate systems or the nodal forces of the cr beam.

**Review by dbaumgaertner — APPROVED** [review]
I am approving this as implementation-wise it looks fine. However, regarding whether or not the last call is necessary, @armingeiser or @MFusseder, could you somehow create a note or project task in git just to make sure this information is not lost (somewhere, where you collect these things for the developments in sensitivity analysis).

**Pull request merged**